polytracker/src/dfsan_pass/DataFlowSanitizer.cpp

```diff
@@ -968,12 +969,19 @@ bool DataFlowSanitizer::runOnModule(Module &M) {
     }
   }
 
+  polytracker::BBSplittingPass bbSplitter;
   for (Function *i : FnsToInstrument) {
     if (!i || i->isDeclaration())
       continue;
 
     removeUnreachableBlocks(*i);
 
+    // Split the function's basic blocks such that each BB has at most one
+    // function call and one conditional branch. This means that all of the
+    // calls in the instrumented code will be followed by an unconditional
+    // branch.
+    bbSplitter.runOnFunction(*i);
```

Comment:
Some more thoughts on this, even though I know it slows down the progress a bit. 

Since we are using opt to load/run passes with polybuild, I think we can write some simple testcases in the form of analysis pass. So we take some code, we generate the bitcode, we run basic block splitter pass on it, 

then our testcase takes that .bc and verifies that each block only has one call or branch statement in it. It would make me feel more confident if we had a testcase or two, I will work on this while you are away on PTO so we can  be closer to merging

Reply:
That's a good idea. I am going to end up closing this PR, but I will create an issue to track this.